recursion.md

```diff
@@ -1,60 +1,61 @@
 
-First we have the file `final_loop_import.scrypt`. This is where you need to define your recursive function, and some utility functions. For us it's very simple : we want to compute the GCD of 2 numbers, so the recursive function will take/return 2 numbers. Because of the type system we need to use an object of a predefined type. This is `Object`. Your recursive function needs to take/return object of type `Object`. While this may look like a limitation, feel free to define 2 struct `Arg` and `Result` so that `Object` will have 2 fields of type `Arg` and `Return`. For GCD we want that `Object` contains 2 ints, because we compute the GCD of 2 numbers.
-Then you need to specify the fee, and the size your structure takes (something similar as `sizeof` in C). A 1byte integer has size 1. We have 2 integers here, so let's take a size of 2.
-And finally you need to write your recursive function, that takes a parameter of type `Object` and returns a type `DoNotTouch`, which has 1boolean which represents if we want to execute the function again, and 1 `Object`, which is the result.
-Please note that you cannot do multiple recursive function calls (btw Fibonacci can be written in a single recursive function call).
-And then at the end of this file you need to write 2 functions, which do the conversion between `bytes` and `Object`. The length of the bytes sequence taken and returned needs to be the size you defined above ! For our example it's simply `num2bin(a, 1) + num2bin(b, 1)`, and a simple parse of a bytes sequence of length 2.
+First we have the file `final_loop_import.scrypt`. This is where you need to define your own recursive function and utility functions. For us it's very simple: we want to compute the GCD of 2 numbers, so the recursive function will take and return 2 numbers, wrapped in a structure called `Object`. Your recursive function needs to take and return object of type `Object`.
 
-Now that we defined our recursive function we need to write our contract ! This is the file `final_loop_main.scrypt`. This is the file of your smart contract. It needs to include two files, and needs to define some public function that you can't touch, but otherwise everything is as usual and you can write the public function you want.
-For our example let's ask the user to provide two numbers which gcd are 1 to spend the utxo. So we need to define a public function that takes two integers a and b, compute gcd, and end with `require (gcd == 1)`.
-The sCrypt code will therefore be `Object result = Loop.computation(txPreImage, {a, b});` to get the GCD, and then `require(result.a == 1)` to make sure it's 1. In our recursive function the GCD is stored on a, that's why it's `result.a`.
+Next you need to specify the fee (satoshi/byte) and the size your structure takes (something similar as `sizeof` in C). One integer has size 1. We have 2 integers here, so the size is 2.
```

Comment:
The fee you need to specify is the fee per transaction, so something like 5000sats, and not of satoshi/byte